R/pkg/R/catalog.R

```diff
@@ -403,21 +406,95 @@ listTables <- function(databaseName = NULL) {
   dataFrame(callJMethod(jdst, "toDF"))
 }
 
+#' Checks if the table with the specified name exists.
+#'
+#' Checks if the table with the specified name exists.
+#'
+#' @param tableName name of the table, allowed to be qualified with catalog name
+#' @rdname tableExists
+#' @name tableExists
+#' @examples
+#' \dontrun{
+#' sparkR.session()
+#' databaseExists("spark_catalog.default.myTable")
+#' }
+#' @note since 3.4.0
+tableExists <- function(tableName) {
+  sparkSession <- getSparkSession()
+  if (class(tableName) != "character") {
+    stop("tableName must be a string.")
+  }
+  catalog <- callJMethod(sparkSession, "catalog")
+  callJMethod(catalog, "tableExists", tableName)
+}
+
+#' Get the table with the specified name
+#'
+#' Get the table with the specified name
+#'
+#' @param tableName the qualified or unqualified name that designates a table, allowed to be
+#'                  qualified with catalog name
+#' @return A data.frame.
+#' @rdname getTable
+#' @name getTable
+#' @examples
+#' \dontrun{
+#' sparkR.session()
+#' tbl <- getTable("spark_catalog.default.myTable")
+#' }
+#' @note since 3.4.0
+getTable <- function(tableName) {
+  sparkSession <- getSparkSession()
+  if (class(tableName) != "character") {
+    stop("tableName must be a string.")
+  }
+  catalog <- callJMethod(sparkSession, "catalog")
+  jtbl <- handledCallJMethod(catalog, "getTable", tableName)
+
+  ret <- data.frame("name" = callJMethod(jtbl, "name"))
+  jcata <- callJMethod(jtbl, "catalog")
+  if (is.null(jcata)) {
+    ret$catalog <- "NULL"
+  } else {
+    ret$catalog <- jcata
+  }
+
+  jns <- callJMethod(jtbl, "namespace")
+  if (is.null(jns)) {
+    ret$namespace <- "NULL"
+  } else {
+    ret$namespace <- paste(jns, collapse = ".")
```

Comment:
`ret$namespace` does not accept a list, otherwise it will raise something like `arguments imply differing number of rows: 1, 0 ...`

here convert namespace to a string

Reply:
Hm,  you should probably assign `list(list(...))`

Reply:
because `ret$namespace <- list(...)` attempts to assign each element of `list` as a whole column. e.g.)

```r
> df <- data.frame(col1 = c('A', 'B', 'C', 'D', 'E'),
+                  col2 = c(1, 2, 3, 4, 5))
>
> df$col3 <- list(1.0, 5.8, 3.0, 5.6, 8.8)
>
> df
  col1 col2 col3
1    A    1    1
2    B    2  5.8
3    C    3    3
4    D    4  5.6
5    E    5  8.8
```

Reply:
it seems that then the frame will be multi-rows:

```R
> data.frame("name" = "tblName", "namespace" = paste(c("ns1", "ns2"), collapse="."))
     name namespace
1 tblName   ns1.ns2
>
> 
> data.frame("name" = "tblName", "namespace" = c(c("ns1", "ns2")))
     name namespace
1 tblName       ns1
2 tblName       ns2
>
> data.frame("name" = "tblName", "namespace" = c(c("ns1"), c("ns2")))
     name namespace
1 tblName       ns1
2 tblName       ns2
> 
>
```

Reply:
Can we work around by direct assignment? e.g.)

```r
> df <- data.frame("name" = "tblName")
> df$namespace <- list(list("ns1", "ns2"))
> df$namespace
[[1]]
[[1]][[1]]
[1] "ns1"

[[1]][[2]]
[1] "ns2"


> df$namespace[[1]][[1]]
[1] "ns1"
> df$namespace[[1]][[2]]
[1] "ns2"
```